ariadne/cardano/test/Test/Ariadne/Bip44.hs

```diff
@@ -0,0 +1,153 @@
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet
+       ( HdAddressChain (..)
+       , HdAccountIx (..)
+       , HdAddressIx(..)
+       , HdCoinType(..))
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Derivation (deriveBip44KeyPairUnwrapped)
+import Cardano.Crypto.Wallet (generate, toXPub, xprv, xpub, unXPrv)
+
+import Pos.Crypto.HD (firstNonHardened, firstHardened)
+import Pos.Core (Address(..), IsBootstrapEraAddr(..), makePubKeyHdwAddress)
+
+import Data.Default (def)
+import Data.ByteArray (convert)
+import qualified Pos.Crypto as S
+import qualified Data.ByteString.Base16 as B16
+
+instance Arbitrary Bip44DerivationPath where
+    arbitrary = do
+      bip44AccountIndex <- arbitrary
+      bip44AddressChain <- arbitrary
+      bip44AddressIndex <- arbitrary
+      bip44CoinType     <- arbitrary
+      return Bip44DerivationPath {..}
+
+{-------------------------------------------------------------------------------
+  Constants
+-------------------------------------------------------------------------------}
+
+-- | BIP-44 derivation constant
+bip44Purpose :: Word31
+bip44Purpose = unsafeMkWord31 44
+
+{-------------------------------------------------------------------------------
+  Check keys derivation
+-------------------------------------------------------------------------------}
+
+-- | Test vector. Contains seed, passphrase, path and reference (derived) private and public keys.
+data TestVector = TestVector { seed :: ByteString, passPhrase :: ByteString, path :: [Word32]
+                             , xprvString :: ByteString, xpubString :: ByteString}
+
+testVectors :: [TestVector]
+testVectors = [ TestVector (fst $ B16.decode $ "58202ed4c71d91bc68c7b50feeb5bc7a785fe884dd0aeddce029df3d612cd3680fd3")
+               ""
+               [2147483648, 2147483649, 2147483650, 2147483650, 3147483648]
+               (fst $ B16.decode $ "624b47150f58dfa44284fbc63c9f99b9b79f808c4955a461f0e2be44eb0be50d097aa006d694b165ef37cf23562e5967c96e49255d2f20faae478dee83aa5b020588589cd9b51dfc028cf225674069cbe52e0e70deb02dc45b79b26ee3548b0015c450b86dd7dd83b31951d9ee03eb1a7925161d817bd517c69cf09e3671f1ca")
+               (fst $ B16.decode $ "0588589cd9b51dfc028cf225674069cbe52e0e70deb02dc45b79b26ee3548b0015c450b86dd7dd83b31951d9ee03eb1a7925161d817bd517c69cf09e3671f1ca")
+              ]
+
+scryptParams :: S.ScryptParams
+scryptParams = fromMaybe (error "failed to make scrypt params through builder") $ S.mkScryptParams def
+
+testEncryptedPass :: S.EncryptedPass
+testEncryptedPass = S.encryptPassWithSalt scryptParams S.emptySalt ("" :: ByteString)
+
+testPassPhrase :: S.PassPhrase
+testPassPhrase = S.PassPhrase $ convert ("" :: ByteString)
+
+genBip44KeyPair :: S.EncryptedSecretKey -> [Word32] -> Maybe (Address, S.EncryptedSecretKey)
+genBip44KeyPair secretKey path = deriveBip44KeyPairUnwrapped (IsBootstrapEraAddr False)
+                                   testPassPhrase secretKey path

```

Comment:
Please avoid this kind of indentation. Add a linebreak after `=` and indent the rest accordingly. Also, you may consider applying eta reduction.